pallets/collator-assignment/src/tests_with_core_config.rs

```diff
@@ -0,0 +1,530 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use crate::assignment::ChainNumCollators;
+use crate::mock::*;
+use crate::{CoreAllocationConfiguration, Pallet};
+use sp_runtime::Perbill;
+use tp_traits::ParaId;
+
+fn generate_parachains_and_parathreads(
+    num_parachains: u32,
+    num_parathreds: u32,
+) -> (Vec<ChainNumCollators>, Vec<ChainNumCollators>) {
+    let mut parachains = vec![];
+
+    for i in 0..num_parachains {
+        parachains.push(ChainNumCollators {
+            para_id: ParaId::new(i + 1),
+            min_collators: 0,
+            max_collators: 0,
+        });
+    }
+
+    let mut parathreads = vec![];
+    for i in 0..num_parathreds {
+        parathreads.push(ChainNumCollators {
+            para_id: ParaId::new(num_parachains + i + 1),
+            min_collators: 0,
+            max_collators: 0,
+        });
+    }
+
+    (parachains, parathreads)
+}
+
+#[test]
+fn test_combine_paras_with_core_config() {
+    let core_count = 10u32;
+    let num_parachains = 7u32;
+    let num_parathreads = 5u32;
+
+    let (generated_parachains, generated_parathreads) =
+        generate_parachains_and_parathreads(num_parachains, num_parathreads);
+
+    // Table for testing
+    // (config, tipping_chains, expected_resulting_chains, expected_to_be_sorted_by_tip, num_collators)
```

Comment:
I like this idea, but can you make it more compact? Maybe have a `function cnc(Vec<u32>) -> Vec<ChainNumCollators>` to avoid all the repeated `min_collators: 0`

Reply:
*fixed*